conans/test/functional/build_requires/test_build_require_invalid.py

```diff
@@ -0,0 +1,317 @@
+                        raise ConanInvalidConfiguration("br cannot be built in debug mode")
+
+                def package_id(self):
+                    del self.info.settings.compiler
+                    if self.settings.build_type == "Debug":
+                        compatible_pkg = self.info.clone()
+                        compatible_pkg.settings.build_type = "Release"
+                        self.compatible_packages.append(compatible_pkg)
+            """)
+    conanfile_consumer = textwrap.dedent("""
+        from conans import ConanFile
+
+        class Conan(ConanFile):
+            name = "libA"
+            version = "1.0.0"
+            settings = "os", "compiler", "build_type", "arch"
+            build_requires = "br/1.0.0"
+        """)
+    client.save({"br_validate.py": conanfile_validate,
+                 "br_configure.py": conanfile_configure,
+                 "br_build.py": conanfile_build,
+                 "br_validate_remove_build_type.py": conanfile_validate_remove_build_type,
+                 "br_configure_remove_build_type.py": conanfile_configure_remove_build_type,
+                 "br_build_remove_build_type.py": conanfile_build_remove_build_type,
+                 "br_validate_compatible.py": conanfile_validate_compatible,
+                 "br_configure_compatible.py": conanfile_configure_compatible,
+                 "br_build_compatible.py": conanfile_build_compatible,
+                 "consumer.py": conanfile_consumer})
+    return client
+
+
+def test_create_invalid_validate(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_validate.py")
+    client.run("create consumer.py -s build_type=Debug", assert_error=True)
+
+
+def test_create_invalid_configure(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_configure.py")
+    client.run("create consumer.py -s build_type=Debug", assert_error=True)
+
+
+def test_create_invalid_build(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_build.py")
+    client.run("create consumer.py -s build_type=Debug", assert_error=True)
+
+
+def test_create_validate_remove_build_type(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_validate_remove_build_type.py")
+    # FIXME: This should NOT fail with -> br/1.0.0: Invalid ID: br cannot be built in debug mode
+    client.run("create consumer.py -s build_type=Debug", assert_error=True)
+
+
+def test_create_configure_remove_build_type(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_configure_remove_build_type.py")
+    # FIXME: This should NOT fail with -> ERROR: br/1.0.0: Invalid configuration: br cannot be built in debug mode
+    client.run("create consumer.py -s build_type=Debug", assert_error=True)
```

Comment:
When using one profile, this scenario is really important too. Many checks are being raised from `configure` method, like the ones related to `cppstd` ([CMake](https://github.com/conan-io/conan-center-index/blob/master/recipes/cmake/3.x.x/conanfile.py)), which are then removed together with the `settings.compiler`.

With this error, it is not possible to use `CMake` as a build-requires and use explicit `-s compiler.cppstd=98`

Reply:
If I understood it correctly, the idea here is to move invalid checks from ``configure()`` to ``validate()``. I am not considering adding extra logic to ``configure()`` step exceptions.

Reply:
Yes, probably **all** the checks can be moved to `validate()` and focus the issue/fix on that function. Maybe this is something we can add explicitly somewhere in ConanCenter docs. Right now, I cannot think about any reason why someone could need to raise from `configure()` instead of `validate()`.